### DISCUSSION TITLE
Power supply via MIC 3385 broken

### DISCUSSION THREAD

**Discussion by CsFreiber** [microchip-pic-avr-solutions/avr-iot-cellular-arduino-library#20] [Q&A]
Hi Simen and others,

we have a problem with one of our IoT Cellular mini moduls. When powering up (via USB or VMUX Jumper) the MCU doesn't react. After measuring the power supply for the MCU we detected a too low voltage ( ca. 0.5V) and the MIC3385 heats up. Therefore we assume this voltage regulator is broken. Are there any other possibilities to power the MCU without the MIC3385? There are no external Pins for power supply without through the regulator.
Have somebody had the same problem and can tell what could caused the defect of the MIC3385?
Thank you for your help!
Best regards

CsFreiber

**Comment by M65649** [comment]
Hello @CsFreiber,

Could you explain more in detail what happened when this occurred? Could you see if forcing the regulator into LDO makes the MCU boot properly? Try to set PB4 logically high. There is an exposed pad on the back of the board in the upper-right corner of the label under D12. 

![Capture](https://github.com/microchip-pic-avr-solutions/avr-iot-cellular-arduino-library/assets/109146295/66429bd2-8bec-41aa-909c-66f9462a4755)

**Reply by CsFreiber** [reply]
Hi @M65649 aka Simen,

thank you for your very quick answer! The problem accured while my colleague played with the low_power example. However I don't know exactly how.
I have tried your suggestion. At first only holding a jumper cable to the pad. The current changed from the external power supply (from max output current at 900mA to ca 600mA), so then I tried to solder a cable to the pad in order to have a constant HIGH (3,3V) connection. Of course I was not successfull. I tried by holding the cable to the pad again while trying to compile a programm. Unfortunatly the following error code occurs

An error occurred while uploading the sketch
avrdude: usbhid_open(): No device found
avrdude: jtag3_open_common(): Did not find any device matching VID 0x03eb and PID list: 0x2175

Now I can not tell, if the pad (PB4 Pin) is surely on HIGH, due to my bad soldering skill. If you have any other suggestions that could help, let me know. Of course you could just power the MCU through the cut strapes.

Best regards

CsFreiber

**Reply by M65649** [reply]
I think you need to open up a support ticket and then we can ship you a new kit. Could you apply here: https://microchip.my.site.com/s/supportservice/?

**Reply by CsFreiber** [reply]
Thank you for your help! We are going to

**Comment by CsFreiber** [comment]
Hi, 
unfortunatly we think, that the _powerDown(const uint32_t power_down_time_seconds)_ function from LowPower class has a bug.
We tested this function on another AVR-IoT-Cellular mini and we had the same issue as with the one in the post above. The MIC 3385 is getting hot and the MCU can not be reseted or newly flashed. By putting the PB4 PIN on logic HIGH did not solve the problem, that we are unable to flash de MCU. Here is a snipet of our code used:

```
void setup() {
  LedCtrl.begin();
  LedCtrl.startupCycle();
  SerialMon.begin(115200);
  smartWI_startup(); //  Start LTE and MQTT connection
}

void loop() {
  LowPower.powerDownModem();
  LowPower.powerDownCPU(10);
  LowPower.restartModem();

  smartWI_startup();

  (modem.testAT()) ? SerialMon.println(F("modem connection OK")) : SerialMon.println(F("modem connection failed"));
  (modem.isNetworkConnected()) ? SerialMon.println(F("Network connected")) : SerialMon.println(F("Network not connected"));
  (mqttclient.connected()) ? SerialMon.println(F("MQTT connection successful")) : SerialMon.println(F("MQTT connection failed!"));

  delay(5000);
}
```

_In the low_power.cpp_

```
void powerDownModem(void){
 SequansController.writeCommand("AT+SQNSSHDN");
 delay(2000);
 Lte.end();
}
```

Basically it is the power_down.ino example from the AVR-IoT-Cellular library. The _powerDownModem()_ function shuts down the modem with the respective AT command. The _powerDownCPU(10)_ is basically the same as the _powerDown()_ function but without calling the _Lte.end()_ and the _while(!Lte.begin())_, as the modem has been already shut down.

However the power_down.ino example seemed to function.
Do you have any idea, what causes this problem? We suggest the PB4 Pin is somehow not disabled again. This can only be the case, when in the _powerDownCPU_ (a.k.a _powerDown()_ ) the cpu doesn't wake up.

**Reply by M65649** [reply]
Hello @CsFreiber,

I’m currently out of office and won’t be back until the 26th of October. If you have any more boards which you are testing on, try to not enable the LDO (you can see the functions enable/disable in low_power.cpp). We have a theory that something is up with that switch in the regulator.

**Reply by CsFreiber** [reply]
Hi @M65649,

thank you so much for your quick supports! We are going to have other AVR boards so no problem. We have also the theory of either the _disableLDO()_ is ill definded or the regulator as you say has an issue. We are looking forward for your answer and eventual solution.
Best regards

CsFreiber

**Reply by M65649** [reply]
No problem, I would advise to either leave LDO out of the picture at the moment or not use low power at all until we can figure out what’s going on.

**Reply by M65649** [reply]
Hello @CsFreiber,

Could you provide some more details about the setup when the fault occurred? Did you have the board connected to anything, did it happen specifically when the device entered low power, or was something else going on?

I'm currently trying to reproduce the error.

**Reply by CsFreiber** [reply]
Hi @M65649 ,

from the setup was nothing special, we just have connected the board via an USB-C cable to our laptop. This was for program and for power up  the board. The code we have used for flashing is as in my post above. After flashing this code the problem occured with the two boards identically. By cutting the power for the MCU by the VMUX Jumper, we could program one of the boards via drag and drop of an other program in HEX file format. By putting the VMUX jumper back, the MCU is programed (the program disables the LDO by the _disableLDO()_ function and just makes the _LEDCtrl.startupCycle()_). However the LEDs are dimmed and the LDO (MIC 3385) is getting hot again. This is suggesting, that the LDO is still in the low power mode and the MCU and the LEDs only getting the 1.8V (this explains the dimmed light from the LEDs). However it does not explain to us, why it is getting hot and why it can not be disabled fully again.
I hope I could explain the setup in a way it is traceable. Feel free to contact me if you need any other detail.

Best regards

CsFreiber

P.S: If the general low power functions are functioning without enableing the LDO, this would be also sufficient for the beginning for our purpose

**Reply by M65649** [reply]
If you are not using the LDO, the power consumption will be far greater in low power. Could you show me the code in the functions `LowPower.powerDownCPU(10);` and `LowPower.restartModem();`?

You were saying that this did not occur when using the `power_down.ino` example, but happened once you tested with your own code?

For the second board you tested with, did this happen instantly/the first time you flashed your own code?

**Reply by CsFreiber** [reply]
Hi @M65649,

here are our two defined functions:
```

void LowPowerClass::restartModem(void) {
  SerialMon.println(F("begin modem restart"));
  digitalWrite(RESET_PIN, HIGH);
  delay(10);
  digitalWrite(RESET_PIN, LOW);
  delay(10);
  SerialMon.println(F("modem restarted"));
  //(modem.testAT()) ? SerialMon.println(F("modem connection OK")) : SerialMon.println(F("modem connection failed"));
}
 
void LowPowerClass::powerDownCPU(const uint32_t power_down_time_seconds) {
 
    SLPCTRL.CTRLA |= SLPCTRL_SMODE_PDOWN_gc | SLPCTRL_SEN_bm;
 
    enablePIT();
    enableLDO();
 
    // Stop millis interrupt to prevent the timer waking up the core
    stop_millis();
 
    uint32_t remaining_time_seconds = power_down_time_seconds;
 
    while (remaining_time_seconds > 0) {
 
        sleep_cpu();
 
        if (pit_triggered) {
            remaining_time_seconds -= 1;
            pit_triggered = false;
        } else {
            // External interrupt caused the CPU to wake
            break;
        }
    }
 
    restart_millis();
 
    disableLDO();
    disablePIT();
    SLPCTRL.CTRLA &= ~SLPCTRL_SEN_bm;
 
    powerUpPeripherals();
}
```

As you can see, the LowPowerCPU is the same function as the _powerDown()_ function without calling the _Lte.end()_ function and the powerDownPeripherals(false). These functions were also defined in the low_power class

We tried both the power_save.ino and power_down.ino examples first, and there was everything fine. Then we wanted a function which only handles the CPU without the modem. The reason is, the modem needs some time to reconnect to the network after Lte.end() was called.

The _restartModem()_ sends a restart pulse to the restart pin. (See the datasheet of Sequans Monarch2 page 14 and AVR-IoT schematic)

The _smartWI_startup()_ is a function which calls Lte.begin(), Lte.isConnected, Mqtt.begin() functions, so also nothing special and nothing with low power or LDO.

In our opinion, the functions we defined has no difference regarding LDO. Also Lte.end() only disconnects the modem, and should not be dependent to the MCU state or LDO. Also the problem occured with both of the boards instantly after we have flashed them.

I hope, we are wrong and you can say, where we made a big mistake. Otherwise we hope you find a bug, which can be fixed.
We thank you again for your help!

Best regards

CsFreiber

**Reply by M65649** [reply]
Thank you, do you call `LowPower.configurePowerDown();` somewhere in your code?

**Reply by CsFreiber** [reply]
Hi,

we think at first we haven't called the `LowPower.configurePowerDown()` function. But we have also tested it, by calling it in our `powerDownModem()` function. There the modem seemed to succesfully shut down.
Even if not, in the `LowPower.configurePowerDown()` you are only calling functions for the Sequans Modem, and it should be independable from our problem with the LDO

Best regards
CsFreiber

**Reply by M65649** [reply]
I did a factory reset of a test board now and tested with the following sketch (removing `Lte.end()` and `Lte.begin()` from `LowPower.powerDown()`). It should be quite equivalent to your setup:


```c
#include <avr/io.h>
#include <led_ctrl.h>
#include <log.h>
#include <low_power.h>
#include <lte.h>
#include <mqtt_client.h>
#include <sequans_controller.h>

#define MQTT_THING_NAME "someuniquemchp"
#define MQTT_BROKER     "test.mosquitto.org"
#define MQTT_PUB_TOPIC  "testtopic"
#define MQTT_PORT       1883
#define MQTT_USE_TLS    false
#define MQTT_USE_ECC    false
#define MQTT_KEEPALIVE  180

#define RESET_PIN PIN_PC5

void smartWI_startup() {
    if (!Lte.begin()) {
        Log.error("Failed to start modem...");
    } else {
        Log.info("Modem running");
    }

    if (!MqttClient.begin(MQTT_THING_NAME,
                          MQTT_BROKER,
                          MQTT_PORT,
                          MQTT_USE_TLS,
                          MQTT_KEEPALIVE,
                          MQTT_USE_ECC)) {
        Log.error("Failed to start MQTT...");
    } else {
        Log.info("MQTT connected");
    }
}

void powerDownModem(void) {
    const ResponseResult response = SequansController.writeCommand(
        "AT+SQNSSHDN");

    if (response != ResponseResult::OK) {
        Log.error("Failed to write shutdown command");
    }

    delay(2000);
    Lte.end();
}

void restartModem(void) {
    digitalWrite(RESET_PIN, HIGH);
    delay(10);
    digitalWrite(RESET_PIN, LOW);
    delay(10);
}

void setup() {
    LedCtrl.begin();
    LedCtrl.startupCycle();

    Log.begin(115200);

    smartWI_startup();
}

void loop() {
    powerDownModem();
    LowPower.powerDown(10);
    restartModem();

    smartWI_startup();

    const ResponseResult response = SequansController.writeCommand("AT");

    if (response != ResponseResult::OK) {
        Log.error("Failed to AT");
    }

    delay(5000);
}
```

This resulted in the following current measured over VMUX:

![Capture](https://github.com/microchip-pic-avr-solutions/avr-iot-cellular-arduino-library/assets/109146295/df686a77-496d-4c5f-86f2-9514593af42d)

I haven't been able to break the regulator (yet) with this setup.

**Reply by CsFreiber** [reply]
Hallo Simen,

thank you for your efforts! Sorry for my late answer, I was not working. This is very strange. At first site you did the same as we did. At this moment I do not have any idea, why this happened with us twice. However we keep investigate the issue and let you know, if we found something.

Best regards

CsFreiber

**Reply by M65649** [reply]
Please let me know if you find anything.

**Reply by CsFreiber** [reply]
Hello Simen,

I just wanted to let you know, that we haven't found the cause of this problem. At our current setup we do not need the LDO for power supply. So for the sleep modus we ignore the functions, which manipulate the LDO. Maybe in the future we have got some time and can further investigate, but at this moment it is not necessary for us and therfore has no priority.
We thank you for your support and hope I/we can ask you any other upcoming questions.

Best regards

CsFreiber

**Reply by M65649** [reply]
Thanks for the update. We haven't been able to find anything either. It's somewhat unfortunate as you get far lower power consumption with the LDO.

Please ask if there's anything else.

**Comment by CsFreiber** [comment]
Hi Simen,

unfortunatly we have experienced again the same issue on an other AVR-IoT-Cellular device. This time we have only used your **power_save** example. We have made only the change to comment out the `enableLDO()` and `disableLDO()` functions in the `LowPowerClass::powerDown,` as we have made bad experiences with it (see mentioned above). The library version of AVR-IoT-Cellular is 1.3.9, of the DxCore is 1.5.11. Sadly the MIC3385 for the power managment gets really hot, so we immediatly cutted the power supply (was via an USB C cable to the laptop). 
The seriel  number of the board is: A09-3437/06   sn1800001232

This time the board seems to function well (the sketch is running and you can flash it), however we haven't powered it for a long time, as we don't want to damage it completly. I have tried your suggestion with connecting the PIN on the back to the GND, but it didn't help. I have uploaded other sketches (for example **debug_modem.ino**), which doesn't require the low power functions, but the MIC gets extremly hot no matter what.

At this moment we don't think it is our software bug, as we haven't used our code but the one from the example. But as the problem occurs even with an other software, maybe it is really a hardware problem.

I asssume you haven't experienced this issue yet, but maybe someone has? Regardless of trying new software out, the problem should be solved, when you flash a code from the examples (as the configurations should be alright). We are really helpless, especially our student, which is trying the low power functions as it is part of his bachelor thesis.

I know it is a burden, but can you show again, if you get a board in the same bad situation as our already 4 or something boards?
If you need more information let me know and thank you for your help!

Best regards

CsFreiber

**Reply by M65649** [reply]
Hello @CsFreiber,

It seems like this is some hardware bug indeed. I have a couple of questions to try to replicate your setup even further:

* Are there any external components connected to the board? Clicks, probes, etc.
* Have you always been powering via USB? When this has occurred, do you know if there has been any ESD?
* Are you probing the board somewhere, in other words, could there be a mismatch between some ground plane somewhere due to external components?
* If you have a broken board that you can't get alive again, have you tried strapping the PB4 pad both to ground and to VCC to see if there is any difference between the switching mode and the LDO regulator? (or possibly cutting the PB4 trace completely to see if that makes any change, but please only attempt this if you have a non-functioning board and you have tested the other options first).

Moreover, have you gotten any support from the official Microchip support yet? It would be beneficial for us if you could send us one (or more) of the broken boards so that we can investigate further.

**Reply by CsFreiber** [reply]
Hello Simen,

sorry for my late answer, I had to disscuss the whole cirumstances with my collegue. Here are the answers for your questions:

- We didn't have any other devices connected to the board as this issue occured
- We have only powered this board via the USB. We are in our Digital.Lab and we have no known ESD issues
- We are probing the boards at our ESD workplace
- We have tested the PB4 pin with connecting to the board internal GND or VCC but there is no change to the LDO. The regulator is getting hot in a few seconds

I have contacted the Microchip support (Jeevitha from the support team), the last time you suggested it. They have only wanted to know the circumstances and said we should contact our distributor, to get a replacement.  This happend and we got a new board. 
However this requires to contact our company purchaser. This results to more costs, then simply buy new devices.
The final answer from the support was: "However, it's important to highlight that the issue concerning the low-power example remains unresolved. The root cause of this problem is currently unclear to us. Therefore, we kindly advise against attempting to use the GitHub code on your new board."
However the thesis of our student is about power managment and the optimization our IoT device for energy consumtion. That's the reason why we are trying the examples out.

I hope I could answer all your question. We could send you the broken board to you, if you like, for further investigation.

Best regards

CsFreiber

**Reply by M65649** [reply]
Hello,

We will contact the team that has designed the regulator to see if we can get more information from them and to hear if they have experienced this before.

Best,
Simen

**Reply by CsFreiber** [reply]
Hello Simen,

thank you so much for your quick support and help! We hope we can contribute in this way a design failure or bug so this do not happen to someone else. We are looking forward for your further news 

Best regards

CsFreiber

**Reply by M65649** [reply]
Hello again @CsFreiber. I just want to let you know that we are still working on this but that things are moving a bit slow.

**Reply by CsFreiber** [reply]
Hello Simen,

thank you for your Update! It is not urgent, as we do not use the LDO at the moment. If you are going to have some news and solutions or if we can help you in some way, let me know.

Best regards

CsFreiber
